filter.h

```diff
@@ -0,0 +1,176 @@
+#ifndef _FILTER_H_
+#define _FILTER_H_
+
+//Standard Includes 
+#include <stdint.h>
+
+
+
+/** @file
+ * @defgroup MovingAvg Filter Module   
+ * 	@ingroup Filters
+ * 
+ * 
+ * This module implements a Filter using a linear transfer function that takes in scale/offset in order to calculate new values  
+ * new_value = raw_value * scale + offset 
+ * 
+ *
+ * This module defaults to a MovingAvg Filter function. The user is resposible for implementing a filter based on application. 
+ * It can be changed using the Set_Filter function 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * Example for implementation with ADC module -- in progress 
+ * @code
+ * 
+ * 
+ * ADC_Init();
+ * Filter_Init();
+ * 
+ * QueueMeasurement()
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * 
+ * @endcode
+ * 
+ * 
+ * 
+ * @author Neil Surti
+ * @author Alexander Revolous 
+ * 
+ * 
+ * **/ 
+
+
+
+
+
+
+
+
+/**
+ *  Filter_Init initliazes all variables/arrays for the filter i.e (min,max,index,) including setting the call back functions to NULL 
+ *  sets the scale/offset values from user and calculates the value for the shift  
+ * 	Sets the default filter to MovingAvgFilter
+ * 
+ * @param scale - Value that multiples with the input 
+ * @param offset - A constant value added to the linear transfer function to compute the desired result if needed
+ * 
+**/
+
+
+void Filter_Init(filter_t *, int16_t  offset, float scale); 
```

Comment:
the typedef for filter_t must be added before it is used otherwise the compiler will cry about it.